packages/react/src/errorboundary.tsx

```diff
@@ -0,0 +1,109 @@
+import * as Sentry from '@sentry/browser';
+import * as hoistNonReactStatic from 'hoist-non-react-statics';
+import * as React from 'react';
+
+export const FALLBACK_ERR_MESSAGE = 'No fallback component has been set';
+export const UNKNOWN_COMPONENT = 'unknown';
+
+export type ErrorBoundaryProps = {
+  showDialog?: boolean;
+  dialogOptions?: Sentry.ReportDialogOptions;
+  fallback?: React.ReactNode;
+  fallbackRender?(error: Error | null, componentStack: string | null, resetErrorBoundary: () => void): React.ReactNode;
+  onError?(error: Error, componentStack: string): void;
+  onMount?(): void;
+  onReset?(error: Error | null, componentStack: string | null): void;
+  onUnmount?(error: Error | null, componentStack: string | null): void;
+};
+
+type ErrorBoundaryState = {
+  error: Error | null;
+  componentStack: string | null;
+};
+
+const INITIAL_STATE = {
+  componentStack: null,
+  error: null,
+};
+
+class ErrorBoundary extends React.Component<ErrorBoundaryProps, ErrorBoundaryState> {
+  public state: ErrorBoundaryState = INITIAL_STATE;
+
+  public componentDidCatch(error: Error, { componentStack }: React.ErrorInfo): void {
+    Sentry.withScope(scope => {
+      scope.setExtra('componentStack', componentStack);
+      Sentry.captureException(error);
+    });
```

Comment:
Try this, new API :)
```suggestion
      Sentry.captureException(error, {contexts: {componentStack}});
```